tools/clang/test/HLSLFileCheck/hlsl/template/DependentWithBuiltinTemplate.hlsl

```diff
@@ -0,0 +1,121 @@
+// RUN: %dxc -E main -T ps_6_0 -enable-templates %s | FileCheck -check-prefixes=CHECK,BUFFER %s
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 %s | FileCheck -check-prefixes=CHECK,BUFFER %s
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DERANGE1 %s | FileCheck %s -check-prefix=ERANGE1
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DERANGE2 %s | FileCheck %s -check-prefix=ERANGE2
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DESCALAR %s | FileCheck %s -check-prefix=ESCALAR
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DEPARTIAL %s | FileCheck %s -check-prefix=EPARTIAL
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DWRAPCB=1 %s | FileCheck -check-prefixes=CHECK,BUFFER %s
+
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DWRAPCB=2 %s | FileCheck -check-prefixes=CHECK,WRAPPER %s
+// RUN: %dxc -E main -T ps_6_0 -HV 2021 -DWRAPCB=3 %s | FileCheck -check-prefixes=CHECK,WRAPPER %s
+
+// BUFFER: %CB0 = type { %"struct.Wrapper<Foo>" }
+// BUFFER: %"struct.Wrapper<Foo>" = type { %struct.Foo }
+
+// WRAPPER: %"hostlayout.$Globals" = type { %"hostlayout.struct.WrappedCB<Foo>" }
+// WRAPPER: %"hostlayout.struct.WrappedCB<Foo>" = type {}
+
+// CHECK: define void @main
+
+// BUFFER: !6 = !{i32 0, %CB0* undef, !"CB0", i32 0, i32 0, i32 1, i32 4, null}
+// WRAPPER: !6 = !{i32 0, %"hostlayout.$Globals"* undef, !"$Globals", i32 0, i32 0, i32 1, i32 0, null}
```

Comment:
Since `WrappedCB` has no data other than the `ConstantBuffer<>` inside, declaring a global of this type should not actually result in any `$Global` cbuffer use.  I think the fact that this is found is an indication of a bug.  As you can see from `%"hostlayout.struct.WrappedCB<Foo>" = type {}`, the buffer type is empty (as it should be), so ultimately, it should not have been included in the cbuffer resource list at all.  Instead, we should have a cbuffer type that indicates the layout of the ConstantBuffer inside the structure, which appears to be missing, if I'm reading this right.  This isn't necessarily something to fix right here, but the test case is checking for something that should not actually be there, and not checking for something that should, if we were to fix the bug.

I will check what the rest of the DXIL/metadata looks like for this case.